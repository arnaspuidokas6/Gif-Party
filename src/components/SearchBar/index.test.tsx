import React from 'react';
import { render, screen } from '@testing-library/react';
import { ERROR_SEARCH, SearchBar } from '.';
import userEvent from '@testing-library/user-event';

describe('SearchBar', () => {
    it('should render', () => {
        // eslint-disable-next-line @typescript-eslint/no-empty-function
        const { container } = render(<SearchBar setSearchValue={() => {}} />);

        expect(screen.getByTestId('search-input')).toBeVisible();
        expect(container).toMatchSnapshot();
    });

    test('if validation works', () => {
        // eslint-disable-next-line @typescript-eslint/no-empty-function
        render(<SearchBar setSearchValue={() => {}} />);
        expect(screen.queryByTestId('invalid-value-message')).toBeNull();

        userEvent.type(screen.getByTestId('search-input'), '++++');
        expect(screen.getByTestId('invalid-value-message')).toHaveTextContent(ERROR_SEARCH);
    });
});
