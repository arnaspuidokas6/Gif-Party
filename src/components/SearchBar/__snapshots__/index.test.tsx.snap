// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar should render 1`] = `
<div>
  <div
    class="box pt-6"
  >
    <div
      class="box-wrapper"
    >
      <div
        class="bg-white rounded flex items-center w-full p-3 shadow-sm border border-gray-200"
      >
        <input
          class="w-full pl-4 text-sm outline-none focus:outline-none bg-transparent"
          data-testid="search-input"
          name="query"
          placeholder="Search for gifs"
          type="search"
        />
      </div>
    </div>
  </div>
</div>
`;
